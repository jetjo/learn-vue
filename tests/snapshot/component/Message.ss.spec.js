import { describe, test, it, expect } from "vitest";
import { mount } from "@vue/test-utils";

import Message from "../../components/Message.vue";

describe("Learn SnapShot", () => {
	it("display message snapshot", () => {
		const wrapper = mount(Message, {
			props: {
				msg: "jetjo",
			},
		});

		// 如果最新的测试中, 结果和上次的快照不同,并且
		// 这种不同是正常的,可以通过给vitest传递u参数来更新快照
		expect(wrapper.text()).toMatchSnapshot();
		expect(wrapper.html()).toMatchSnapshot();
		// 这种方案可以保持快照对于的语言的语法高亮
		expect(wrapper.html()).toMatchFileSnapshot('./__snapshots__/Message.ss.spec.snap.html');
	});
});