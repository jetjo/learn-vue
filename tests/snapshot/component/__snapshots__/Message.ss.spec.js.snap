// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Learn SnapShot > display message snapshot 1`] = `"jetjo"`;

exports[`Learn SnapShot > display message snapshot 2`] = `"<div>jetjo</div>"`;
